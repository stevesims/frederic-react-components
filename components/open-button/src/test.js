import React from 'react';
import { shallow } from 'enzyme';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

import Component from '.';

expect.extend(createMatchers(emotion));

describe('OpenButton', () => {
  let wrapper;

  it('renders with no props', () => {
    wrapper = shallow(<Component />);
    expect(wrapper.find('RotatingButton')).toHaveStyleRule('border', 'none');
  });
    
  it('renders with open prop', () => {
    wrapper = shallow(<Component open />);
    expect(wrapper.find('RotatingButton')).toHaveStyleRule('transform', 'rotate(0deg)');
  });

  it('calls onChange event when clicked', () => {
    const handleClick = jest.fn();
    wrapper = shallow(<Component onChange={handleClick} />);
    wrapper.simulate('click');
    expect(handleClick).toHaveBeenCalledWith({open: true});
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

