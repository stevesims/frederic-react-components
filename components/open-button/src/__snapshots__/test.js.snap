// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenButton matches snapshot 1`] = `
<RotatingButton
  onClick={[Function]}
>
  <Arrow
    fill="#0C60A2"
    width={20}
  />
</RotatingButton>
`;
